nucypher/blockchain/eth/sol/source/contracts/PolicyManager.sol

```diff
@@ -154,6 +174,19 @@ contract PolicyManager is Upgradeable {
         nodeInfo.previousFeePeriod = _period;
     }
 
+    /**
+    * @notice Migrate from the old period length to the new one
+    * @param _node Node address
+    */
+    function migrate(address _node) external onlyEscrowContract {
+        NodeInfo storage nodeInfo = nodes[_node];
+        // with previous period length any previousFeePeriod will be greater than current period
+        // this is a sign of not migrated node
+        require(nodeInfo.previousFeePeriod >= getCurrentPeriod());
+        nodeInfo.previousFeePeriod = recalculatePeriod(nodeInfo.previousFeePeriod);
```

Comment:
Ok, so if a node has reached period 1400, but the new protocol is on period 200, then clearly they haven't yet migrated. When they do migrate,  the `recalculatePeriod` function converts to the correct period? 

This makes sense intuitively, but the genesis protocol began with period 18550, so I expected a more complicated calculation than simply multiplying the `previousFeePeriod` by `genesisSecondsPerPeriod / secondsPerPeriod` (= 1/7). Unless on the migration date both the new and old protocol will reset the period count to 0?

Reply:
But this is really simple, just divide and that's it. In some moments we use idea that all genesis periods more than all new periods.